### PR TITLE
add-aws-cni-annotations

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/14441

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Review by pipo02mix — APPROVED** [review]
LGTM but I would use IP as it is an acronym too

**Review by paurosello — APPROVED** [review]
LGTM

**Review by njuettner — APPROVED** [review]

File: pkg/annotation/aws.go [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
 
+// AWSCNIMinimumIPTarget is the annotation to configure value for MINIMUM_IP_TARGET for AWS CNI
+// see https://github.com/aws/amazon-vpc-cni-k8s#cni-configuration-variables
+// and https://github.com/aws/amazon-vpc-cni-k8s/blob/master/docs/eni-and-ip-target.md
+const AWSCNIMinimumIPTarget = "alpha.aws.giantswarm.io/cni-minimum-ip-target"
+
+// AWSCNIWarmIPTarget is the annotation to configure value for WARM_IP_TARGET for AWS CNI
+// see https://github.com/aws/amazon-vpc-cni-k8s#cni-configuration-variables
+// and https://github.com/aws/amazon-vpc-cni-k8s/blob/master/docs/eni-and-ip-target.md
+const AWSCNIWarmIPTarget = "alpha.aws.giantswarm.io/cni-warm-ip-target"
```
Comment (xh3b4sd) [inline pkg/annotation/aws.go:11]: What about this structure? I feel like this is a more appropriate structure for the domain namespace. 

```suggestion
const AWSCNIWarmIPTarget = "cni.alpha.aws.giantswarm.io/warm-ip-target"
```

Comment (calvix) [reply pkg/annotation/aws.go:11]: @giantswarm/sig-ux @marians  WDYT?

File: pkg/annotation/aws.go [hunk 2]
```diff
@@ -1,5 +1,15 @@
 package annotation
 
+// AWSCNIMinimumIPTarget is the annotation to configure value for MINIMUM_IP_TARGET for AWS CNI
```
Comment (marians) [inline pkg/annotation/aws.go:3]:
```suggestion
// AWSCNIMinimumIPTarget is an annotation to configure the value for MINIMUM_IP_TARGET for AWS CNI
```

File: pkg/annotation/aws.go [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+// see https://github.com/aws/amazon-vpc-cni-k8s#cni-configuration-variables
+// and https://github.com/aws/amazon-vpc-cni-k8s/blob/master/docs/eni-and-ip-target.md
+const AWSCNIMinimumIPTarget = "alpha.aws.giantswarm.io/cni-minimum-ip-target"
+
+// AWSCNIWarmIPTarget is the annotation to configure value for WARM_IP_TARGET for AWS CNI
```
Comment (marians) [inline pkg/annotation/aws.go:8]:
```suggestion
// AWSCNIWarmIPTarget is an annotation to configure the value for WARM_IP_TARGET for AWS CNI
```

File: pkg/annotation/aws.go (hunk 1 above)
Comment (marians) [reply pkg/annotation/aws.go:11]: I don't think we have system regarding annotation grouping, and I can't judge if we need this group.

Just noting, the prefix could be either of

- `cni.alpha.aws.giantswarm.io/`
- `alpha.cni.aws.giantswarm.io/`

Comment (calvix) [reply pkg/annotation/aws.go:11]: `alpha.cni.aws.giantswarm.io/` would make more sense for me

Comment (calvix) [reply pkg/annotation/aws.go:11]: but uncertain since we don't use it anywhere else so far

Comment (marians) [reply pkg/annotation/aws.go:11]: Then I vote for `alpha.cni.aws.giantswarm.io/`, too

**Comment by calvix** [comment]
RFR with the changes to the annotation structure

**Review by xh3b4sd — APPROVED** [review]

**Pull request merged**